view/src/components/DiffViewWrapper.jsx

```diff
@@ -152,27 +231,31 @@ export default function DiffViewWrapper({
     };
 
     setHunks(file.hunks);
+    // If showing code for onlyOneSide, then right now we are showing the whole function.
+    if (onlyOneSide) return;
     addFirstAndLastLine();
     setHunks((oldHunks) => getHunksInRange(oldHunks));
-    // Showing function which is called instead of diff of function
-    if (linesToShow) {
-      if (isFunctionLong(oldStart, oldEnd)) {
-        // old calling line
-        addLineWithContext(linesToShow[0]);
-        // add new calling line in case if it is in different change
-        // than old calling line
-        const oldLineOfNewCalling = getCorrespondingOldLineNumber(
-          file.hunks,
-          linesToShow[1],
-        );
-        if (oldLineOfNewCalling !== linesToShow[0]) {
-          addLineWithContext(oldLineOfNewCalling);
-        }
-      } else {
-        addAllLinesOfFunction();
+    if (showDiff) return;
```

Comment:
Why is this here?

Reply:
Good question. This line was added in `Viewer: Show entire definition for macros` commit,  I realize now that the commit message did not fully describe the change I made.  The intention was to keep the behavior for macros containing differences unchanged -- by default, showing only the lines with syntax differences (+ context lines).
If there is a difference, the lines with syntax differences are already shown (respectively included in `hunks` to be shown later), the  lines below `    if (showDiff) return;` determine other lines of source code which should be shown. In case of macros all lines would be added. So by this line, I am ending the function early to ensure that we show only the differing lines and their context.

I updated the commit message so it includes the information that for differing macros we are by default showing only the syntax differences. Also, I added a comment above the line, so it is more clear why it is there.

Reply:
Ah, ok, makes sense now. Thanks!